site/docs/how-to-release.md

```diff
@@ -270,6 +270,46 @@ git tag -am 'Release Apache Iceberg <VERSION>' apache-iceberg-<VERSION> apache-i
 
 Then release the candidate repository in [Nexus](https://repository.apache.org/#stagingRepositories).
 
+#### Update docs and javadoc for the release
```

Comment:
There is already a section called "### Documentation Release" can we move it under that?
It points to a readme and maybe we should update these steps in that readme.

Reply:
Reason I put here is because imho, it should be part of the release publication and should be mentioned as part of the release process.
No problem to update, but release manager has to go there (which is not obvious imho).

Reply:
I agree with Ajantha that it looks like there are two sections about docs:

https://github.com/apache/iceberg/blob/47220768d16b9e3b43af48c0e0d92b30aba43c09/site/docs/how-to-release.md#documentation-release

The one that's already there doesn't add much value since the publication steps are not in there

Reply:
@ajantha-bhat resuming this PR: I guess you are proposing to move update docs and javadoc in the `documentation-release` section right ?

Reply:
Maybe we can even move these info in the README pointed out in that section. 

We just need all docs related stuffs in one place.

Reply:
I agree, let me rework this. Thanks.